src/main/java/com/minecraftabnormals/endergetic/core/mixin/PlayerListMixin.java

```diff
@@ -8,20 +8,33 @@
 import net.minecraft.entity.EntityType;
 import net.minecraft.entity.player.ServerPlayerEntity;
 import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.CompressedStreamTools;
 import net.minecraft.nbt.ListNBT;
 import net.minecraft.network.NetworkManager;
+import net.minecraft.server.MinecraftServer;
 import net.minecraft.server.management.PlayerList;
+import net.minecraft.util.registry.DynamicRegistries;
 import net.minecraft.world.server.ServerWorld;
+import net.minecraft.world.storage.PlayerData;
 import net.minecraftforge.fml.network.PacketDistributor;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.injection.At;
 import org.spongepowered.asm.mixin.injection.Inject;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 
+import java.io.File;
 import java.util.List;
 
 @Mixin(PlayerList.class)
 public final class PlayerListMixin {
+	private PlayerData playerDataManager;
+	private MinecraftServer server;
+
+	@Inject(at = @At("RETURN"), method = "<init>")
+	private void initialize(MinecraftServer p_i231425_1_, DynamicRegistries.Impl p_i231425_2_, PlayerData p_i231425_3_, int p_i231425_4_, CallbackInfo info) {
+		playerDataManager = p_i231425_3_;
+		server = p_i231425_1_;
+	}
```

Comment:
This way of setting `playerDataManager` and `server` from the `PlayerList` is unnecessary.
Use `@Shadow` and `@Final` to set `playerDataManager` and `server` with their values.